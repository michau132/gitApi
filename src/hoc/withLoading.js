import React from 'react';
import styled, { keyframes } from 'styled-components';

const spin = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const LoadingSpinner = styled.div`
  ${({ loadingPosition }) => (loadingPosition
    ? 'position: absolute; bottom: 0'
    : 'position: fixed; top: 40%'
  )};
  left: 46%;
  transform: translate(-50%, -50%);
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid ${({ theme }) => theme.color.title};
  width: 120px;
  height: 120px;
  animation: ${spin} 2s linear infinite;
`;

const LoadingWrapper = styled.div`
  position: relative;
`;

const withLoading = Component => (props) => {
  const {
    loading, beerList, beer,
  } = props;
  // checking if the modal is rendering
  if (beer && loading) {
    return <LoadingSpinner />;
  }

  // positioning spinner to infinite scroll and if beerList have data make position to bottom: 0;
  let loadingPosition;
  if (beerList && beerList.length > 0) {
    loadingPosition = true;
  }
  return (
    <LoadingWrapper>
      <Component {...props} />
      {loading && <LoadingSpinner loadingPosition={loadingPosition} />}
    </LoadingWrapper>
  );
};

export default withLoading;
