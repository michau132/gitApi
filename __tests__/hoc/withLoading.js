import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withLoading from '../../src/hoc/withLoading';

const component = () => <h1>Hi there</h1>;
const ConditionalHOC = withLoading(component);
const props = { loading: true };

describe('testing withLoading HOC', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ConditionalHOC {...props} />);
  });
  test('should render component when loading is set to true', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
