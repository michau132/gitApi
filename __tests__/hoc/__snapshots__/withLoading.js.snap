// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing withLoading HOC should match snapshot 1`] = `
<styled.div>
  <component
    loading={true}
  />
  <styled.div />
</styled.div>
`;
